opensafely/__init__.py

```diff
@@ -27,35 +28,19 @@ def show_help(**kwargs):
     parser_help = subparsers.add_parser("help", help="Show this help message and exit")
     parser_help.set_defaults(function=show_help)
 
-    # Add `run` subcommand
-    parser_run = subparsers.add_parser("run", help=local_run.DESCRIPTION)
-    parser_run.set_defaults(function=local_run.main)
-    local_run.add_arguments(parser_run)
+    def add_subcommand(cmd, module):
+        subparser = subparsers.add_parser(cmd, help=local_run.DESCRIPTION)
+        subparser.set_defaults(function=module.main)
+        handles_unknown = module.add_arguments(subparser)
+        if handles_unknown:
+            supports_unknown_args.add(module.main)
```

Comment:
I think there might be a neater way of doing this which is to call:
```py
subparser.set_defaults(handles_unknown_args=False)
module.add_arguments(subparser)
```

And then if the module wants to opt in to looser parsing its `add_arguments` function can call:
```py
subparser.set_defaults(handles_unknown_args=True)
```

Then there's no need to track this in a set, and no need to have the boolean return value from `add_arguments`.

Reply:
Ah nice, will do